plugin/src/main/java/org/opensearch/ml/action/profile/MLProfileTransportAction.java

```diff
@@ -103,27 +123,104 @@ private MLProfileNodeResponse createMLProfileNodeResponse(MLProfileRequest mlPro
-                log.debug("Runtime task profile is found for model {}", mlTask.getModelId());
-                mlLocalTasks.put(taskId, mlTask);
-            }
-        });
-        Arrays.stream(mlModelManager.getAllModelIds()).forEach(modelId -> {
-            if (mlProfileInput.isReturnAllModels() || (!mlProfileInput.emptyModels() && targetModelIds.contains(modelId))) {
-                log.debug("Runtime model profile is found for model {}", modelId);
-                MLModelProfile modelProfile = mlModelManager.getModelProfile(modelId);
-                if (modelProfile != null) {
-                    mlLocalModels.put(modelId, modelProfile);
+
+        CountDownLatch latch = new CountDownLatch(1);
+        boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
+        searchHiddenModels(ActionListener.wrap(hiddenModels -> {
+            Arrays.stream(mlTaskManager.getAllTaskIds()).forEach(taskId -> {
+                MLTask mlTask = mlTaskManager.getMLTask(taskId);
+                if (isSuperAdmin || !hiddenModels.contains(mlTask.getModelId())) {
+                    if (mlProfileInput.isReturnAllTasks()
+                        || (!mlProfileInput.emptyTasks() && mlProfileInput.getTaskIds().contains(taskId))) {
+                        log.debug("Runtime task profile is found for model {}", mlTask.getModelId());
+                        mlLocalTasks.put(taskId, mlTask);
+                    }
+                    if (mlProfileInput.isReturnAllTasks()
+                        || (!mlProfileInput.emptyModels() && targetModelIds.contains(mlTask.getModelId()))) {
+                        log.debug("Runtime task profile is found for model {}", mlTask.getModelId());
+                        mlLocalTasks.put(taskId, mlTask);
+                    }
+                }
+            });
+            Arrays.stream(mlModelManager.getAllModelIds()).forEach(modelId -> {
+                if (isSuperAdmin || !hiddenModels.contains(modelId)) {
+                    if (mlProfileInput.isReturnAllModels() || (!mlProfileInput.emptyModels() && targetModelIds.contains(modelId))) {
+                        log.debug("Runtime model profile is found for model {}", modelId);
+                        MLModelProfile modelProfile = mlModelManager.getModelProfile(modelId);
+                        if (modelProfile != null) {
+                            if (isSuperAdmin && hiddenModels.contains(modelId)) {
+                                modelProfile.setIsHidden(Boolean.TRUE);
+                            }
+                            mlLocalModels.put(modelId, modelProfile);
+                        }
+                    }
                 }
-            }
-        });
+            });
+        }, e -> { log.error("Search Hidden model wasn't successful"); }), latch);
+
+        try {
+            latch.await(10, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            // Handle interruption if necessary
```

Comment:
Still return MLProfileNodeResponse for this node with empty results or a flag in the response showing this node exists in the cluster?

Reply:
it will return with a empty result.

Reply:
I mean returning "node-id": { models: ..., tasks: ... } rather than returning {}, so the Profile API results show the nodeId so we know the node still exists in the cluster.

Reply:
If any model isn't deployed or there isn't any task, currently it shows `{}`, so I don't think I'm breaking any CX experience.

Reply:
The difference is that what would be the implication of {} after this change? Is it expired or nothing in the memory at all?